carcara/src/checker/rules/pb_blasting.rs

```diff
@@ -375,14 +375,92 @@ pub fn pbblast_pbbconst(RuleArgs { conclusion, .. }: RuleArgs) -> RuleResult {
     Ok(())
 }
 
+/// Helper to transform a bitvector to a list of terms, both when short-circuited or not
+/// Ex: `get_bitvector_terms(x, 2)`
+/// >>> `[((int_of 0) x),((int_of 1) x)]`
+///
+/// Ex: `get_bitvector_terms((pbbterm @x0 @x1), 2)`
+/// >>> `[@x0, @x1]`
+fn get_bitvector_terms(bv: &Rc<Term>, pool: &mut dyn TermPool) -> Vec<Rc<Term>> {
+    if let Some(xs) = match_term!((pbbterm ...) = bv) {
+        xs.to_vec()
+    } else {
+        // Get bit width of `x`
+        let n = get_bit_width(bv, pool).unwrap();
+        (0..n)
+            .map(|i| {
+                build_term!(
+                    pool,
+                    ((_ int_of { pool.add(Term::new_int(i)) }) { bv.clone() })
+                )
+            })
+            .collect()
+    }
+}
+
 /// Implements the bitwise exclusive or operation.
 pub fn pbblast_bvxor(RuleArgs { .. }: RuleArgs) -> RuleResult {
     Err(CheckerError::Unspecified)
 }
 
 /// Implements the bitwise and operation.
-pub fn pbblast_bvand(RuleArgs { .. }: RuleArgs) -> RuleResult {
-    Err(CheckerError::Unspecified)
+pub fn pbblast_bvand(RuleArgs { pool, conclusion, .. }: RuleArgs) -> RuleResult {
+    let ((x, y), bit_constraints) =
+        match_term_err!((= (bvand x y) (pbbterm ...)) = &conclusion[0])?;
+
+    let xs = get_bitvector_terms(x, pool);
+    let ys = get_bitvector_terms(y, pool);
+
+    // Zip three lists into tuples
+    for ((bc, xi), yi) in bit_constraints.iter().zip(xs.iter()).zip(ys.iter()) {
+        let (bindings, (c1, c2, c3)) = match_term_err!((choice ... (and c1 c2 c3)) = bc)?;
+
+        // Single binding
+        assert!(bindings.len() == 1);
+        // Check z -> Int
+        let (z_name, z_type) = &bindings[0];
+        assert!(z_name == "z");
+        assert!(*z_type.as_sort().unwrap() == Sort::Int);
+
+        // c1 : (>= @x0 z)
+        let (xic, zc) = match_term_err!((>= xi z) = c1)?;
+        rassert!(
+            xic == xi,
+            CheckerError::TermEquality(EqualityError::ExpectedEqual(xic.clone(), xi.clone()))
+        );
```

Comment:
Here, you can use the [`assert_eq` function](https://github.com/ufmg-smite/carcara/blob/a7888530241f85a9cfa849fa0f5e1b9047bb0fc2/carcara/src/checker/rules/mod.rs#L104-L113) from the `rules` module:
```suggestion
        super::assert_eq(xic, xi)?;
```
(or `assert_is_expected`, whichever you think is more appropriate -- the only difference is the error message)